### ISSUE TITLE
NullPointerException in setBitmapForLockscreen()'s FetchBitmapTask.onPostExecute()

### ISSUE THREAD

**Issue by obec** [googlecast/CastCompanionLibrary-android#233]
```
java.lang.NullPointerException: Attempt to invoke virtual method 'android.support.v4.media.session.MediaControllerCompat android.support.v4.media.session.MediaSessionCompat.getController()' on a null object reference
	at com.google.android.libraries.cast.companionlibrary.cast.VideoCastManager$29.onPostExecute(VideoCastManager.java:2287)
	at com.google.android.libraries.cast.companionlibrary.cast.VideoCastManager$29.onPostExecute(VideoCastManager.java:2284)
	at android.os.AsyncTask.finish(AsyncTask.java:636)
	at android.os.AsyncTask.access$500(AsyncTask.java:177)
	at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:653)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5254)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

More details on reproducing from PR #214 :
> When Clicking on Stop Casting while the Bitmap Lock Screen Runs its fetch bitmap task, the Application tries to Access getController().getMetadata(). However since we already disconnected, getController() will return null and result in a crash.

**Comment by naddaf** [comment]
Could you please create a new stack trace since the line numbers do not match. Also, please provide the complete steps to reproduce (e.g. does it matter what page you are on, etc).

**Comment by obec** [comment]
The line numbers appear to match in the above stack trace, but not in the PR. 
> at com.google.android.libraries.cast.companionlibrary.cast.VideoCastManager$29.onPostExecute(VideoCastManager.java:2287)
https://github.com/googlecast/CastCompanionLibrary-android/blob/master/src/com/google/android/libraries/cast/companionlibrary/cast/VideoCastManager.java#L2287

I can see if there's a consistent way to trigger this.

**Comment by obec** [comment]
The easiest way I've found to reproduce the issue is to proxy a device to throttle its connection.

Steps:
1. Proxy a device with Charles or similar to throttle its connection.
2. Connect to Cast device with media playing, or begin a new cast session.
3. As the lock screen image is loading over the network, disconnect from the cast device.

As soon as the image finishes downloading a NPE will be thrown.

Looking through VideoCastManager if clearMediaSession is called while the lock screen image is still downloading, there's a potential for the NPE to be thrown. For example: 

1. Begin casting a single video with Sender A.
2. Proxy a second device, Sender B, with Charles or similar to throttle its connection.
3. When the media from Sender A is close to finishing, connect to the Cast device with Sender B.

If playback of the media from Sender A completes before the lock screen image request from Sender B, a NullPointerException will be thrown.

**Comment by naddaf** [comment]
Should be fixed in 2.5.2
